### PR TITLE
Implement create and get vehicles

### PR DESCRIPTION
**Pull request by PatrickCmd**
### What does this PR do?
- Enable admin user to add vehicles.
- Enable users to view vehicles list.

#### Description of Task to be completed?
- Implement endpoints for creating and retrieving vehicles
- Test all endpoints and edge cases.

#### How should this be manually tested?
- get the branch with `git fetch origin add_vehicle_endpoint`
- checkout the branch `git checkout add_vehicle_endpoint`
- Run `docker-compose build` and `docker-compose up -d` to start server
- on the terminal type the command `docker exec -it ridemyway-api-django_app_1 python manage.py createsuperuser` and follow the prompts to create superuser
- make a login post request to api/v1/auth/users/login/ with the superuser's credentials
```
{
       "email": "user-email",
        "password": "user-password"
}
```
- set authorization headers with the returned token
- make a post request to /api/v1/vehicles/ to create a vehicle
```
{
	"number_plate": "car number plate",
         "category": "CR",
         "color": "car color",  // this is an optional field with options `CR` or `BK`
         "make": "car make",
	"model": "car model"
}
```
- make a get request to /api/v1/vehicles/ to get a list of vehicles

#### Any background context you want to provide?
- N/A

#### What are the relevant pivotal tracker stories?
- #163976363 #163977331

#### Screenshots(if applicable):
Login user
Screenshot 2019-09-24 at 18 24 05

Create a vehicle
Screenshot 2019-09-24 at 18 24 32

Get vehicle list
Screenshot 2019-09-24 at 18 25 05

### CODE REVIEW

File: ridemyway/tests/base_test_case.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+
+class VehicleBaseTestCase(BaseTestCase):
+    """A base test class to setup vehicle reusable for integration testing
+    with vehicle endpoints.
+    """
+    def setUp(self):
+        super().setUp()
+        self.vehicle = self.setup_vehicle()
+        self.vehicle_uri = "/api/v1/vehicles/"
```
Comment (marthamareal) [inline ridemyway/tests/base_test_case.py]: You can use `reserve` and the name of the url, in case of any changes in the path, there won’t be need of making changes to different places.

File: ridemyway/tests/integration/vehicle/test_vehicle.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
+from ridemyway.tests.base_test_case import VehicleBaseTestCase
+
+
+class TestVehicle(VehicleBaseTestCase):
+    def test_create_vehicle(self):
+        login_uri = "/api/v1/auth/users/login/"
```
Comment (marthamareal) [inline ridemyway/tests/integration/vehicle/test_vehicle.py]: same here

File: ridemyway/tests/integration/vehicle/test_vehicle.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            response.status_code,
+            201,
+            "Expected Response Code 201, received {0} instead.".format(
+                response.status_code
+            )
+        )
+        self.assertIn("Vehicle successfully created", str(response.data))
+    
+    def test_get_vehicle_list(self):
+        login_uri = "/api/v1/auth/users/login/"
```
Comment (marthamareal) [inline ridemyway/tests/integration/vehicle/test_vehicle.py]: And here.

File: ridemyway/tests/base_test_case.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    
+    def setup_vehicle(self):
```
Comment (marthamareal) [inline ridemyway/tests/base_test_case.py:53]: I think since this is in the Base class, we can create a Vehicle instance in the setUp without using the function.
`self.vehicle = VehicleFactory()`

File: ridemyway/tests/integration/vehicle/test_vehicle.py [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    def test_create_vehicle(self):
+        login_uri = "/api/v1/auth/users/login/"
+        params_admin_user = {
+            "email":  self.admin_user.email,
+            "password": "password123"
+        }
+
+        self.set_authorization_header(login_uri, params_admin_user)
+
+        vehicle_params = {
```
Comment (marthamareal) [inline ridemyway/tests/integration/vehicle/test_vehicle.py]: For this data, we have a file under `tests/test_data.py` where we can declare it from and import it, for multiple usages.

File: ridemyway/tests/integration/vehicle/test_vehicle.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
+from ridemyway.tests.base_test_case import VehicleBaseTestCase
+
+
+class TestVehicle(VehicleBaseTestCase):
+    def test_create_vehicle(self):
```
Comment (marthamareal) [inline ridemyway/tests/integration/vehicle/test_vehicle.py:10]: I think we can add on the test names whether it succeeds or fails eg `test_create_vehicle_succeeds`

**Review by marthamareal — CHANGES_REQUESTED** [review]
Greate work @PatrickCmd

**Comment by marthamareal** [comment]
@PatrickCmd we can also add some validations to data input on creating vehicle

File: ridemyway/tests/base_test_case.py (hunk 1 above)
Comment (PatrickCmd) [reply ridemyway/tests/base_test_case.py]: I see makes sense.

File: ridemyway/tests/base_test_case.py (hunk 4 above)
Comment (PatrickCmd) [reply ridemyway/tests/base_test_case.py:53]: I will change it.

File: ridemyway/tests/integration/vehicle/test_vehicle.py (hunk 6 above)
Comment (PatrickCmd) [reply ridemyway/tests/integration/vehicle/test_vehicle.py:10]: Will add test cases with invalid or empty fields.

File: ridemyway/tests/integration/vehicle/test_vehicle.py (hunk 2 above)
Comment (PatrickCmd) [reply ridemyway/tests/integration/vehicle/test_vehicle.py]: Done

File: ridemyway/tests/integration/vehicle/test_vehicle.py (hunk 5 above)
Comment (PatrickCmd) [reply ridemyway/tests/integration/vehicle/test_vehicle.py]: Done

**Comment by PatrickCmd** [comment]
@marthamareal can you please point out the validations we can add for vehicle creation.

**Review by marthamareal — APPROVED** [review]
Great work
